app/controllers/users/emails_controller.rb

```diff
@@ -62,6 +63,18 @@ def delete
       redirect_to account_url
     end
 
+    def properties
+      ParseControllerFromReferer.new(request.referer).call
+    end
+
+    def cancel_link_url
+      if properties[:request_came_from] == 'sign_up/completions#show'
+        sign_up_completed_url
+      else
+        decorated_sp_session.cancel_link_url
+      end
+    end
+
```

Comment:
Bit of a nitpick, but methods are cheap, and it could help readability to extract a method that's named in a way that describes what we're checking.

```suggestion
      if adding_from_completions_screen?
        sign_up_completed_url
      else
        decorated_sp_session.cancel_link_url
      end
    end

    def adding_from_completions_screen?
      properties[:request_came_from] == 'sign_up/completions#show'
    end
```